subprojects/docs/src/docs/userguide/dependencyLocking.adoc

```diff
@@ -0,0 +1,144 @@
+The validation is as follows:
+
+* Existing entries in the lockfile must be matched in the build
+** A version mismatch or missing resolved module causes a build failure
+* Resolution result must not contain extra dependencies compared to the lock file content
+
+=== Generating and updating lockfiles
+
+In order to generate or update a lockfile, you specify the `--write-locks` command line argument in addition to the normal tasks that would trigger configurations to be resolved.
+This will cause the creation of a lockfile for each resolved configuration in that build execution.
+Note that if a lockfile existed previously, it is overwritten.
+
+==== Lock all configurations in one build execution
+
+When locking multiple configurations, you may want to lock them all at once, during a single build execution.
+
+For this, you have two options:
+
+* Run `./gradlew dependencies --write-locks`.
+This will effectively lock all resolvable configurations that have locking enabled.
+Note that in a multi project setup, `dependencies` only is executed on _one_ project, the root one in this case.
+* Declare a custom task that will resolve all configurations
+++++
+<sample id="locking-all" dir="userguide/dependencyManagement/dependencyLocking/lockingAllConfigurations" title="Resolving all configurations">
+    <sourcefile file="build.gradle" snippet="resolve-all"/>
+</sample>
+++++
+
+=== Controlled lock file update
+
+In order to update only specific modules of a configuration, you can use the `--update-locks` command line flag.
+It takes a comma (`,`) separated list of module notations.
+In this mode, the existing lock file is still used as input to resolution, filtering out the modules targeted by the update.
+Note however that the resolution may cause other module versions to update, as dictated by the Gradle resolution rules.
+
+----
+❯ ./gradlew classes --update-locks org:foo,org:bar
+----
+
+Wildcards, indicated with `*`, can be used in the group or module name. They can be the only character or appear at the end of the group or module respectively.
+The following wildcard notations are valid:
+
+* `org*:foo`
+* `org:*`
+* `*:bar`
+* `org*:bar*`
+
+=== Deleting a lockfile
+
+Simply remove the file matching the configurations where you no longer want locking.
+
+=== Locking limitations
+
+It is currently not possible to lock the `classpath` configuration used for build plugins.
+
+=== Nebula locking plugin
+
+This feature is inspired by the https://github.com/nebula-plugins/gradle-dependency-lock-plugin[Nebula Gradle dependency lock plugin].
+It is effectively an adaptation of the features of that plugin that leverage the recent developments around dependency management in Gradle.
+
+There are differences though between the plugin features and what is integrated into Gradle core.
+
+The main difference is that the scope of locking in Gradle is per configuration, and thus implicitly per project.
```

Comment:
This doesn't explain the difference. What is the scope of locking in the Nebula plugin?

Reply:
My understanding is that there were some other limitations of the plugin that have now been lifted. I hope we can say something about those in whatever migration text we have.

Reply:
Removed